### PR TITLE
Docs: Overhaul the Development Environment section of the Block Editor Handbook

### PR DESCRIPTION
**Pull request by ndiego**
## What?
The PR aims to update the [Development Environment](https://developer.wordpress.org/block-editor/getting-started/devenv/) section of the Block Editor Handbook and provide a more streamlined developer onboarding experience using `wp-env` or `wp-now`.

## Why?
The current section is not up to date and does not provide an optimal developer experience. It also does not cover `wp-now`.

## How?
To begin, the main [Development Environment](https://developer.wordpress.org/block-editor/getting-started/devenv/) page is simplified. It now provides an overview of what you need to set up a dev environment for JavaScript development and provides brief explanations of "why" you need a code editor, Node.js dev tools, and a local WordPress environment. 

The section covering Node.js dev tools is broken out into its own page, providing more details and installation information.

The section on creating a local WordPress environment has also been broken out into two additional guides, one for `wp-env` and another for `wp-now`. Since these are the two tools currently maintained by WordPress itself, this is why they were chosen. This structure allows additional guides to be added in the future depending on community interest. 

Finally, the current [How to setup local WordPress environment on Ubuntu](https://developer.wordpress.org/block-editor/getting-started/devenv/docker-ubuntu/) doc has been removed, and that content is now part of the Troubleshooting section of the new "Guide to using wp-env".

### CODE REVIEW

File: docs/getting-started/devenv/README.md [hunk 1]
```diff
@@ -1,228 +1,72 @@
 # Development Environment
 
-This guide is for setting up your local environment for JavaScript development for creating plugins and tools to extend WordPress and the block editor. If you are looking to contribute to Gutenberg project itself, see additional documentation in the [Getting Started guide](/docs/contributors/code/getting-started-with-code-contribution.md).
+This guide will help you set up a local environment for JavaScript development, which can be used to create blocks and other plugins that extend and modify the Block Editor in WordPress.
 
-A development environment is a catch-all term for what you need setup on your computer to work. The three main pieces needed for our development environment are:
+To contribute to the Gutenberg project itself, refer to the additional documentation in the code contribution [Getting Started](/docs/contributors/code/getting-started-with-code-contribution.md) guide.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: I would avoid the word "Getting Started" again when pointing to the code contribution guide to make this "Getting Started" term resonate more within this section

> To contribute to the Gutenberg project itself, refer to the additional documentation in the [code contribution guide](/docs/contributors/code/getting-started-with-code-contribution.md).

**Comment by juanmaguitar** [comment]
@ndiego I have created the following diagram and shared some videos to share my thoughts.
Let me know what you think. Also happy to jump into a call if you prefer to.

https://excalidraw.com/#json=Y3QKUa0x85sIkUwYAutSA,oej4myPLor2dTxSTnIu_lg

image

https://www.loom.com/share/Part-1-Thoughts-on-Docs-Overhaul-the-Developer-Environment-section-of-the-Block--c1d3ff532be14aa1bdb4fb81572443f5?sid=f89deaa1-606a-49a2-a3fd-ca5062446793

https://www.loom.com/share/Part-2-Thoughts-on-Docs-Overhaul-the-Developer-Environment-section-of-the-Block--cc2d01f075cf440e96ca278330ab1a99?sid=fb89d0c1-6af8-4350-a5b6-b22a3a6c20f5

File: docs/getting-started/devenv/README.md [hunk 2]
```diff
@@ -1,228 +1,51 @@
-# Development Environment
+# Block Development Environment
 
-This guide is for setting up your local environment for JavaScript development for creating plugins and tools to extend WordPress and the block editor. If you are looking to contribute to Gutenberg project itself, see additional documentation in the [Getting Started guide](/docs/contributors/code/getting-started-with-code-contribution.md).
+This guide will help you set up a local environment for JavaScript development, which can be used to create blocks and other plugins that extend and modify the Block Editor in WordPress.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: Is it just a "local environment for JavaScript development"? We also need a PHP interpreter and a MySQL DB in place to develop blocks. I would suggest making this sentence more inclusive/generic: 
> This guide will help you set up the right development environment to create blocks and other plugins that extend and modify the Block Editor in WordPress.

File: docs/getting-started/devenv/README.md [hunk 3]
```diff
@@ -1,228 +1,51 @@
...
 
-A development environment is a catch-all term for what you need setup on your computer to work. The three main pieces needed for our development environment are:
+To contribute to the Gutenberg project itself, refer to the additional documentation in the code contribution [Getting Started](/docs/contributors/code/getting-started-with-code-contribution.md) guide.
 
-1. Node/NPM Development Tools
-2. WordPress Development Site
-3. Code Editor
+A development environment includes the tools you need on your computer to successfully develop for the Block Editor. The three essential components are:
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: I would include the terms "block development environment" and "requirements" (for SEO purposes) in this sentence
> A block development environment includes the tools you need on your computer to successfully develop for the Block Editor. The three essential requirements are:

File: docs/getting-started/devenv/README.md [hunk 4]
```diff
@@ -1,228 +1,51 @@
...
 
-## Quick Start
+1.  Code editor
+2.  Node.js development tools
+3.  Local WordPress environment (site)
 
-Here is a summary of the guide. See each section for additional details and explanations.
+## Choosing a code editor
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: I think we should make these headings match the previous list (that are focused on the "what"). I would just call this title:
> Code editor

File: docs/getting-started/devenv/README.md [hunk 5]
```diff
@@ -1,228 +1,51 @@
...
-**1. Install Node development tools**
+A code editor is used to write code, and you can use whichever editor you're most comfortable with. The key is having a way to open, edit, and save text files.
 
-Download and install [Node Version Manager](https://github.com/nvm-sh/nvm) (nvm)
+If you do not already have a preferred code editor, [Visual Studio Code](https://code.visualstudio.com/) (VS Code) is a popular choice for JavaScript development among Core contributors. It works well across the three major platforms (Windows, Linux, and Mac) and is open-source and actively maintained by Microsoft. VS Code also has a vibrant community providing plugins and extensions, including many for WordPress development.
 
-```
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
-```
+## Installing Node.js development tools
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: As per the previous comment, I would just call this title:
> Node.js development tools

File: docs/getting-started/devenv/README.md [hunk 6]
```diff
@@ -1,228 +1,51 @@
...
+To install Node.js development tools and learn more about each, refer to the links below.
 
-Start the environment from an existing plugin or theme directory, or a new working directory:
+-   [Install Node.js for Mac and Linux](/docs/getting-started/devenv/nodejs-development-tools.md#mac-and-linux-installations)
+-   [Install Node.js for Windows](/docs/getting-started/devenv/nodejs-development-tools.md#windows-or-alternative-installs)
 
-```
-wp-env start
-```
+## Setting up a local WordPress environment
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: As per the previous comment, I would just call this title:
> Local WordPress environment

File: docs/getting-started/devenv/README.md [hunk 7]
```diff
@@ -1,228 +1,51 @@
...
+-   Setup local WordPress environments with `wp-env` and `wp-now`
+-   Use the latest ECMAScript features and write code in ESNext
+-   Lint, format, and test JavaScript code
+-   Scaffold custom blocks with the `create-block` package
 
-First download, install, and start [Docker Desktop](https://www.docker.com/products/docker-desktop) following the instructions for your OS.
+The list goes on. While modern JavaScript development can be challenging, WordPress provides several tools, like [`wp-scripts`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/) and [`create-block`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/), that streamline the process and are made possible by Node.js development tools.
 
--   Install WordPress environment tool
+If you are already familiar with `node`, it's recommended that you use the [LTS](https://nodejs.dev/en/about/releases/) (Long Term Support) version, and you will also need Node Package Manager (`npm`), the `npx` script, and Node Version Manager (`nvm`).
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: I would rephrase this sentence to surface the importance of `nvm` and highlight the recommended Node version (LTS)

> A Node version manager tool like [`nvm`](https://github.com/nvm-sh/nvm) is strongly recommended to easily change your Node version when required. The **recommended Node version for Block Development is the [Active LTS](https://nodejs.dev/en/about/releases/) (Long Term Support) version**. Any Node installation includes tools such as the Node Package Manager (`npm`) and the Node Package eXecute (`npx`) that will be required to work with some of the [WP packages](https://github.com/WordPress/gutenberg/tree/trunk/packages).

File: docs/getting-started/devenv/README.md [hunk 8]
```diff
@@ -1,228 +1,51 @@
...
-If this does not work and the `$NVM_DIR` environment variable is set you can remove using `rm -rf "$NVM_DIR"`
-
-To clean up any installed JavaScript packages remove the global `.npm` directory at `$HOME/.npm`,
-
-```
-rm -rf "$HOME/.npm"
-```
-
-Just as you confirmed the installation worked, you can confirm the uninstall worked. First quit and restart terminal and then try to run `npm -v`, `node -v`, and `nvm -v` you should then see a "command not found" error in the terminal.
+Of the two, `wp-env` is the more robust and complete solution. It's also the recommended tool for Gutenberg development. On the other hand, `wp-now` offers a simplified setup and does not require Docker. Both are valid approaches, so the choice is yours.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: I would avoid mentioning "Docker" for  `wp-now` as it wasn't mentioned as a requirement for `wp-env` and would highlight `wp-env` as the recommended tool. 

I would rephrase this as: 

> Of the two, **`wp-env` is the more solid and complete solution**. It's also the recommended tool for Gutenberg development. On the other hand, `wp-now` offers a simplified setup but is more limited than `wp-env`. Both are valid options, so the choice is yours.

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
+# Node.js development tools
+
+When developing for the Block Editor, you will need [Node.js](https://nodejs.org/en) development tools along with a code editor and a local WordPress environment. Node.js (`node`) is an open-source runtime environment that allows you to execute JavaScript outside of the web browser.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: I would be more explicit about Node in the last sentence of this paragraph:

> Node.js (`node`) is an open-source runtime environment that allows you to execute JavaScript code from the terminal (also known as a command-line interface, CLI, or shell)

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+Installing `node` will automatically include Node Package Manager (`npm`) which includes the script `npx`, two tools you will frequently use in block and plugin development.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: I would rephrase this to mention both aliases and to refer to both of them as tools 

> Installing `node` will automatically include the Node Package Manager (`npm`) and the Node Package eXecute (`npx`), two tools you will frequently use in block and plugin development.

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+Node Package Manager (`npm`) serves multiple purposes, including dependency management and script execution. It's the recommended package manager and is extensively featured in all documentation.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: I would add this link
> [`npm`](https://docs.npmjs.com/cli/v10/commands/npm)

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+The `npx` script is used to run commands from packages without installing them globally and is commonly used when scaffolding blocks with the [`create-block`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/) package.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: I would  rephrase this part and add a link to `npx`
> The  Node Package eXecute [`npx`](https://docs.npmjs.com/cli/v10/commands/npx) tool is used to run commands from packages without installing them globally and is commonly used when scaffolding blocks with the [`create-block`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/) package.

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## Mac and Linux installations
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: I would be more explicit in the title about the content of this section
> ## Node installation on Mac and Linux (with nvm)

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+2. Quit and restart the terminal.
+3. Run `nvm install lts` in the terminal to install the latest [LTS](https://nodejs.dev/en/about/releases/) (Long Term Support) version of Node.js.
+4. Run `node -v` and `npm -v` in the terminal to verify the installed `node` and `npm` versions.
+
+If needed, you can also install specific versions of `node`. For example, install version 18 by running `nvm install 18`, and switch between different versions by running `nvm use [version-number]`. See the `nvm` [usage guide](https://github.com/nvm-sh/nvm#usage) for more details.
+
+Some projects, like Gutenberg, include an [`.nvmrc`](https://github.com/WordPress/gutenberg/blob/trunk/.nvmrc) file which specifies the version of `node` that should be used. In this case, running `nvm use` will automatically select the correct version. If the version is not yet installed, you will get an error that tells you what version needs to be added. Run `nvm install [version-number]` followed by `nvm use`.
+
+## Windows or alternative installs
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: I would rephrase this title to be aligned with the previous suggestion

> ## Node installation on Windows and others

File: docs/getting-started/devenv/README.md [hunk 15]
```diff
@@ -1,228 +1,51 @@
...
 
-1. Node/NPM Development Tools
-2. WordPress Development Site
-3. Code Editor
+A development environment includes the tools you need on your computer to successfully develop for the Block Editor. The three essential components are:
 
-## Quick Start
+1.  Code editor
+2.  Node.js development tools
+3.  Local WordPress environment (site)
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: I would include links to the headings of each subsection below

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+Here are the quick instructions for installing `node` using `nvm`. See the [complete installation guide](https://github.com/nvm-sh/nvm#installing-and-updating) for more details.
+
+1. Open the terminal and run the following to install `nvm`. On macOS, the required developer tools are not installed by default. Install them if prompted.
+
+```sh
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.5/install.sh | bash
+```
+
+2. Quit and restart the terminal.
+3. Run `nvm install lts` in the terminal to install the latest [LTS](https://nodejs.dev/en/about/releases/) (Long Term Support) version of Node.js.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: The actual command is `nvm install --lts` as per [this](https://github.com/nvm-sh/nvm#long-term-support)

**Comment by ndiego** [comment]
Thanks @juanmaguitar, I have addressed all feedback in the most recent commit!

**Comment by juanmaguitar** [comment]
What about changing the name of the folder `docs/getting-started/devenv` to something more self-explanatory such as `docs/getting-started/block-development-environment`? 
This will be also helpful for SEO as the final URL in the handbook for this page would be something like https://developer.wordpress.org/block-editor/getting-started/block-development-environment/

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+Installing `node` will automatically include the Node Package Manager (`npm`) and the Node Package eXecute (`npx`), two tools you will frequently use in block and plugin development.
+
+Node Package Manager ([`npm`](https://docs.npmjs.com/cli/v10/commands/npm)) serves multiple purposes, including dependency management and script execution. It's the recommended package manager and is extensively featured in all documentation.
+
+The Node Package eXecute ([`npx`](https://docs.npmjs.com/cli/v10/commands/npx)) tool is used to run commands from packages without installing them globally and is commonly used when scaffolding blocks with the [`create-block`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/) package.
+
+
+## Node.js installation on Mac and Linux (with `nvm`)
+
+For Mac and Linux, it's recommended that you use [Node Version Manager](https://github.com/nvm-sh/nvm) (`nvm`). This allows you to install and manage specific versions of `node`, which are installed locally in your home directory, avoiding any global permission issues.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/nodejs-development-tools.md]: Maybe we can make this paragraph more generic to also invite Windows users to install nvm (as it's mentioned in the Windows section). What about something like the following?

> The recommended way to install Node.js is via [Node Version Manager](https://github.com/nvm-sh/nvm) (`nvm`). This allows you to [...]

File: docs/getting-started/devenv/README.md [hunk 18]
```diff
@@ -1,228 +1,51 @@
...
+-   Scaffold custom blocks with the `create-block` package
 
-First download, install, and start [Docker Desktop](https://www.docker.com/products/docker-desktop) following the instructions for your OS.
+The list goes on. While modern JavaScript development can be challenging, WordPress provides several tools, like [`wp-scripts`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/) and [`create-block`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/), that streamline the process and are made possible by Node.js development tools.
 
--   Install WordPress environment tool
+The recommended Node.js version for block development is [Active LTS](https://nodejs.dev/en/about/releases/) (Long Term Support). However, there are times when you  need to to use different versions. A Node.js version manager tool like `nvm` is strongly recommended and allows you to easily change your `node` version when required. You will also need Node Package Manager (`npm`) and the Node Package eXecute (`npx`) to work with some WordPress packages. Both are installed automatically with Node.js.
```
Comment (juanmaguitar) [inline docs/getting-started/devenv/README.md]: Could we highlight the following sentence even more?
> The recommended Node.js version for block development is [Active LTS](https://nodejs.dev/en/about/releases/) (Long Term Support).

We could wrap it in a `<div class="callout-tip">` or just make this sentence bold.

**Review by juanmaguitar — APPROVED** [review]
I have added a few comments but preemptively approved this PR so you're not blocked to move forward with it.
